lsif/src/shared/config/config.ts

```
@@ -58,7 +53,13 @@ export async function waitForConfiguration(logger: Logger): Promise<Configuratio
         }).catch(() => {})
     })
 
-    return () => oldConfiguration!
+    return () => {
+        if (oldConfiguration === undefined) {
+            throw new Error('Configuration is not defined.')
```

Can this happen? The only way I can see is if someone calls the function right after `waitForConfiguration`, which is actually done to initialize the tracer. But this error is not handled at that callsite, which would tell me that it actually can't happen, in which case an assertion would be more correct (or rewriting the code, so that the Promise above resolves with the initial config and therefor ensures it is always set here).

---

Updated.